ext_mongodb.php

```diff
@@ -620,6 +626,94 @@ class BulkWriteException extends WriteException {}
 /* {{{ BSON and Serialization Classes */
 namespace MongoDB\BSON;
 
+/* {{{ Interfaces */
+interface BinaryInterface
+{
+	public function serialize() : string;
+	public function unserialize(mixed $data) : void;
```

Comment:
My point is that the PHPC classes cannot satisfy these interfaces, because they only define a `__wakeup()` method and the properties to serialize are obtained by the `get_properties` handler.

Why would you plan to use Serializable in HHVM (i.e. "C" type serialization) instead of relying on `__sleep()` and `__wakeup()` (i.e. "O" type)? That will means each driver uses a different serialization format.

As for the interfaces, I think we can simply omit serialization methods and allow users to decide which serialization method they want to support (if any).